processor/resourcedetectionprocessor/factory.go

```diff
@@ -96,7 +97,7 @@ func (f *factory) createTraceProcessor(
 		return nil, err
 	}
 
-	return newResourceTraceProcessor(ctx, nextConsumer, provider, oCfg.Override), nil
+	return newResourceDetectionTracesProcessor(ctx, nextConsumer, provider, oCfg.Override), nil
```

Comment:
In a future PR, it would be good to update these to use `processorhelper.NewTracesProcessor`, etc. Then we could cleanup the empty `Shutdown` function, etc.

Reply:
That's a great idea, I actually just applied this suggestion. It nicely reduces the boilerplate